lib/main/atom/views/semanticViewPane.ts

```diff
@@ -0,0 +1,120 @@
+import {CompositeDisposable, PaneItemObservedEvent} from "atom"
+import {SemanticView, SEMANTIC_VIEW_URI} from "./semanticView"
+
+export class SemanticViewPane {
+  isOpen: boolean = false
+  subscriptions: CompositeDisposable | null = null
+
+  public activate(state: any) {
+    if (!SEMANTIC_VIEW_URI && state) {
+      // NOTE is is just a dummy to avoid warning of unused variable state
+      console.log(state)
+    }
+    this.subscriptions = new CompositeDisposable()
+
+    this.subscriptions.add(
+      atom.workspace.addOpener((uri: string) => {
+        if (uri === "atom://" + SEMANTIC_VIEW_URI) {
+          this.isOpen = true
+          const view = new SemanticView({})
+          view.start()
+          return view
+        }
+      }),
+    )
+
+    this.subscriptions.add({
+      dispose() {
+        atom.workspace.getPaneItems().forEach(paneItem => {
+          if (paneItem instanceof SemanticView) {
+            paneItem.destroy()
+          }
+        })
+      },
+    })
+
+    this.subscriptions.add(
+      atom.workspace.onDidAddPaneItem((event: PaneItemObservedEvent) => {
```

Comment:
This definitely looks weird. If I closed/opened the semantic view, it doesn't necessarily mean the settings should change, does it? Besides, tracking the events doesn't make all that much sense either, because we're fully in control of when those events happen (via `show`/`hide`/`toggle`).

Same with `onDidDestroyPaneItem`

Reply:
I guess it depends on how you want the view to behave:   
I'd prefer that my last decision would be remembered, when restarting Atom, instead of having to open the settings and change it there.

But that is probably a matter of taste -- there should probably be an additional setting for controlling that, i.e. if the editor should remember my last decision, or use a fixed setting for opening/not opening the semantic view the next time Atom is started(?)

But I definitely move it into the show/hide (or better yet serialization/deserialization hooks) -- I wasn't sure, if Atom has some other way of controlling the view, so the listeners seemed the safest way to catch opening/closing of the view.

Reply:
> I'd prefer that my last decision would be remembered, when restarting Atom, instead of having to open the settings and change it there.

It's not necessarily a decision. I.e. you can't open semantic view *without* changing the settings currently, and that seems weird to me. I mean, I hate it when software implicitly does something I don't necessarily want it to do like silently changes settings that are supposedly user-controlled.

Besides, having it change globally doesn't make much sense either, since whether you want to see it or not depends mostly on the project you're working on. So serializing semantic view visibility state makes much more sense than writing to global config.